### PR TITLE
getTime refactoring

### PR DESCRIPTION
**Pull request by Ekdohibs**
Rebased version of #1188.

### CODE REVIEW

**Comment by celeron55** [comment]
Why are the variables storing getTimeMs() results changed from u32 to u64 when getTimeMs() itself still returns u32?

**Comment by Ekdohibs** [comment]
That's a good question to which I don't have the answer: it was like that in the original pull request. I think I'm going to change the return type of the getTime* functions to u64 then.

**Comment by celeron55** [comment]
You'll also have to edit the functions to do the thing they do in ShadowNinja's PR so that they actually return 64-bit values. Otherwise there's no point.

**Comment by nerzhul** [comment]
is this necessary to have this PR as we will switch to C++11 after release and we will have std::chrono nice features ?

File: src/gettime.h [hunk 1]
```diff
@@ -21,34 +21,156 @@ with this program; if not, write to the Free Software Foundation, Inc.,
...
+		case PRECISION_NANO:
+			return getTimeNs();
+	}
+	return 0;
+}
+
+/**
+ * Delta calculation function taking two 32bit arguments.
+ * @param old_time_ms old time for delta calculation (order is relevant!)
+ * @param new_time_ms new time for delta calculation (order is relevant!)
```
Comment (SmallJoker) [inline src/gettime.h]: Relevant? It gets the delta of it anyways.

**Comment by ShadowNinja** [comment]
You should probably take a look at my patch (3625a31cd2899997d75900a91ee9591c174cfcac).  You might be able to just apply my patch without any of the code from the other patch since they do just about the same thing (except that mine switches to u64 properly).

**Comment by ShadowNinja** [comment]
b662a4577d692329b9ca83525e6039f2ddcd1ac1

**Pull request closed without merging**